server/events/vcs/github_client_test.go

```diff
@@ -1014,3 +1014,86 @@ func TestGithubClient_Retry404(t *testing.T) {
 	Ok(t, err)
 	Equals(t, 3, numCalls)
 }
+
+func TestGithubClient_PullIsLocked(t *testing.T) {
+	respTemplate := `{
+		"state": "pending",
+		"statuses": [
+		  {
+			"url": "https://api.github.com/repos/owner/repo/statuses/832812d4777ddc4197685c5a8f864eaf8a82d4ae",
+			"avatar_url": "https://avatars.githubusercontent.com/in/123528?v=4",
+			"id": 14049059268,
+			"node_id": "MDEzOlN0YXR1c0NvbnRleHQxNDA0OTA1OTI2OA==",
+			"state": "pending",
+			"description": "{\"pr_number\": 176, \"waiting\": %s}",
+			"target_url": "",
+			"context": "sq-ready-to-merge",
+			"created_at": "2021-08-09T19:43:56Z",
+			"updated_at": "2021-08-09T19:43:56Z"
+		  }
+		],
+		"sha": "832812d4777ddc4197685c5a8f864eaf8a82d4ae",
+		"total_count": 1
+	  }`
+
+	/*
+	  1. Locked description.
+	  2. Unlocked description.
+	*/
+
+	cases := []struct {
+		state     string
+		resp      string
+		expLocked bool
+	}{
+		{
+			"locked",
+			fmt.Sprintf(respTemplate, `[\"approval\", \"lock\"]`),
+			true,
+		},
+		{
+			"unlocked",
+			fmt.Sprintf(respTemplate, `[\"approval\"]`),
+			false,
+		},
+	}
```

Comment:
nit: I know atlantis does this a lot, but i truly hate this style of testing.  I like being explicit with inputs and outputs even if the code gets repetitive. Makes the tests more maintainable going forward.

Reply:
Yup, this way of testing was new for me and definitely a little confusing 😅 I thought that's how the team was writing unit tests figures I should just stick with it. But, I'm glad we don't hehe! I've updated the tests.